src/transaction/declare.rs

```diff
@@ -381,7 +381,10 @@ impl Declare {
             Rc<RefCell<ProgramCache<'_, ClassHash>>>,
         >,
     ) -> Result<TransactionExecutionInfo, TransactionError> {
-        if self.version != Felt252::ONE && self.version != Felt252::ZERO {
+        if !(self.version == Felt252::ZERO
+            || self.version == Felt252::ONE
+            || self.version == Felt252::THREE)
+        {
```

Comment:
Declare V3 behaves similar to DeclareV2 rather than the older Declare. We should move this to the DeclareV2 implementation. (We can rename it to DeclareV2Plus after 0.13 changes are done and merged to avoid confusions)

Reply:
Great!, i move the check to DeclareV2 struct and also created the issue
https://github.com/lambdaclass/starknet_in_rust/issues/1208